libs/common/src/main/java/org/opensearch/common/hash/T1ha1.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.hash;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.VarHandle;
+import java.nio.ByteOrder;
+
+import static java.lang.Long.rotateRight;
+
+/**
+ * t1ha: Fast Positive Hash
+ *
+ * <p>
+ * Implements <a href="https://github.com/erthink/t1ha#t1ha1--64-bits-baseline-fast-portable-hash">t1ha1</a>;
+ * a fast portable hash function with reasonable quality for checksums, hash tables, and thin fingerprinting.
+ *
+ * <p>
+ * To overcome language and performance limitations, this implementation differs slightly from the reference
+ * implementation in C++, so the returned values may vary before JDK 18.
```

Comment:
Could you please add the link to reference implementation in C++?

Reply:
Added.